pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -937,13 +1016,24 @@ cdef class _Timedelta(timedelta):
         -----
         Any nanosecond resolution will be lost.
         """
-        return timedelta(microseconds=int(self.value) / 1000)
+        if self._reso == NPY_FR_ns:
+            return timedelta(microseconds=int(self.value) / 1000)
+
+        # TODO(@WillAyd): is this the right way to use components?
+        self._ensure_components()
+        return timedelta(
+            days=self._d, seconds=self._seconds, microseconds=self._microseconds
+        )
 
     def to_timedelta64(self) -> np.timedelta64:
         """
         Return a numpy.timedelta64 object with 'ns' precision.
         """
-        return np.timedelta64(self.value, 'ns')
+        cdef:
+            str abbrev = npy_unit_to_abbrev(self._reso)
+        # TODO: way to create a np.timedelta64 obj with the reso directly
+        #  instead of having to get the abbrev?
```

Comment:
@seberg is there a C-API way to create a timedelta64 object?

Reply:
> assuming you got the correct dtype available

what we have on hand is the correct NPY_DATETIMEUNIT.  I guess we need to create the dtype from the unit (we have a function to go the other direction, so i guess this shouldn't be too hard to figure out).  If I figure this out, I'll probably try to upstream it into numpy's `__init__.pxd`

Reply:
It would probably be OK to add a function that works with the unit directly for the C-API, also.  But it doesn't exist yet.  It seems `PyArray_Scalar` is commented out from `__init__.pxd`, I am not sure if there is a reason for that.

Reply:
could the reason by that `PyArray_Scalar`'s first arg is `void *` which might not play so well with cython?  i've figured out how to create the dtype object from the unit (copied `create_datetime_dtype_with_unit` over from multiarray/datetime.c) but so far having no luck in calling `PyArray_Scalar`

Reply:
Maybe?  I don't really see a good reason, `void *` seems perfectly fine, if cython doesn't like it, just use `char *` (which is likely better anyway?)
But, I also don't think it is API that should be used a lot, so that may just be the reason also, that it is pretty unused.